clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -4385,6 +4386,80 @@ class CIRSignBitOpLowering : public mlir::OpConversionPattern<cir::SignBitOp> {
   }
 };
 
+class CIRInitListCtorLowering
+    : public mlir::OpConversionPattern<cir::InitListCtor> {
+public:
+  using OpConversionPattern<cir::InitListCtor>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(cir::InitListCtor op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    const bool inbounds = true;
+    rewriter.setInsertionPoint(op);
+
+    mlir::Type llvmPtrType =
+        mlir::LLVM::LLVMPointerType::get(rewriter.getContext());
+
+    // initlist object
+    auto initListType = mlir::cast<cir::StructType>(
+        mlir::cast<cir::PointerType>(op->getOperand(0).getType()).getPointee());
+    assert(initListType.getMembers().size() == 2U);
+    auto initListLLVMType = getTypeConverter()->convertType(initListType);
+    mlir::Value initlist = adaptor.getOperands()[0];
+
+    // initlist data array
+    unsigned argListSize = op->getNumOperands() - 1U;
+    assert(argListSize >= 1U);
+
+    mlir::Type llvmElementType = adaptor.getArgList().getTypes()[0];
+#ifndef NDEBUG
+    for (unsigned i = 1; i < argListSize; i++)
+      assert(llvmElementType == adaptor.getArgList().getTypes()[i]);
+#endif
+    mlir::Type llvmArrayType =
+        mlir::LLVM::LLVMArrayType::get(llvmElementType, argListSize);
+    mlir::Value llvmArrayPtr = rewriter.create<mlir::LLVM::AllocaOp>(
```

Comment:
It's important for all the `allocas` to be at the start of the function, otherwise LLVM will generate worse code.